src/shared/components/plots/BoxScatterPlot.tsx

```diff
@@ -561,7 +568,7 @@ export default class BoxScatterPlot<D extends IBaseBoxScatterPlotPoint> extends
         const maxSize = Math.max(
             ...this.labels.map(x=>getTextWidth(x, axisTickLabelStyles.fontFamily, axisTickLabelStyles.fontSize+"px"))
         );
-        if (this.props.horizontal) {
+        if (!!this.props.compressAxisX || this.props.horizontal) {
             // if horizontal mode, its label width
```

Comment:
please add to this comment indicating why the compressAxisX condition is relevant